### PR TITLE
Use proper die message on duplicate job name

### PR DESCRIPTION
**Pull request by okurz**
The 'NAME' field is checked to be unique for jobs and can fail but the value
'0' is not a good choice for that cause. Triggering a single job was plain
wrong assuming the job return value to always be a valid argument which was
then catched and forwarded to the API caller in a totally misleading error
message. For the iso the job creation failure was just ignored in before. Now
the code would just die on the iso post request which should be much more
helpful.

### CODE REVIEW

**Comment by Martchus** [comment]
Good, I was already wondering why the error is not handled here. Unfortunately you have to resolve conflicts with my latest changes.

File: lib/OpenQA/Schema/ResultSet/Jobs.pm [hunk 1]
```diff
@@ -114,11 +114,11 @@ sub create_from_settings {
...
     my %new_job_args = (TEST => $settings{TEST});
 
-    if ($settings{NAME}) {
-        my $njobs = $self->search({slug => $settings{NAME}})->count;
-        return 0 if $njobs;
 
-        $new_job_args{slug} = $settings{NAME};
+    if (my $name = $settings{NAME}) {
+        my $njobs = $self->search({slug => $name})->count;
+        !$njobs or die "job with name \'$name\' already exists in database";
```
Comment (Martchus) [inline lib/OpenQA/Schema/ResultSet/Jobs.pm]: What speaks against adding a unique constraint on database level?

Comment (okurz) [inline lib/OpenQA/Schema/ResultSet/Jobs.pm]: "slug" is already a unique constraint. What speaks against removing whole "slug" is the question

**Comment by okurz** [comment]
rebased

File: lib/OpenQA/WebAPI/Controller/API/V1/Iso.pm [hunk 2]
```diff
@@ -360,19 +360,17 @@ sub schedule_iso {
 
             # create a new job with these parameters and count if successful, do not send job notifies yet
             my $job = $self->db->resultset('Jobs')->create_from_settings($settings);
+            push @jobs, $job;
 
-            if ($job) {
```
Comment (okurz) [inline lib/OpenQA/WebAPI/Controller/API/V1/Iso.pm]: question to reviewers: Should we keep the behaviour to silently ignore if a job can not be created or catch with eval?

**Comment by coolo** [comment]
I don't think we ever used the slug - the idea there was to have it linkable by slug instead of ID, because the cgi version of openQA did it this way.

But IMO we shouldn't allow the NAME or slug to be set from the outside at all.

**Comment by okurz** [comment]
should I just ignore `$settings{NAME}`?

**Comment by coolo** [comment]
I think we should. NAME is generated - but the last time I tried to change something I noticed we actually have a test case for that and dropped it for that moment :)

**Comment by coolo** [comment]
So how to continue?

File: t/04-scheduler.t [hunk 3]
```diff
@@ -158,6 +159,10 @@ $settings2{NAME}  = "OTHER NAME";
 $settings2{BUILD} = "44";
 my $job2 = $schema->resultset('Jobs')->create_from_settings(\%settings2);
 
+# try to call a
+throws_ok { $schema->resultset('Jobs')->create_from_settings(\%settings2) }
+qr/job with name \'OTHER NAME\' already exists in database/, 'No two jobs with same name accepted';
```
Comment (kraih) [inline t/04-scheduler.t]: There's no need to escape the apostrophe.

**Comment by okurz** [comment]
obsoleted by https://github.com/os-autoinst/openQA/pull/1109

**Pull request closed without merging**